### PR TITLE
Participation websocket service - follow up

### PR DESCRIPTION
**Pull request by SiggZ**
### Checklist
- [x] I run `yarn run webpack:build:main`: the project builds without errors.
- [x] I run `yarn lint`: the project builds without code style warnings.
- [x] ~I updated the documentation and models.~
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] ~I added (end-to-end) test cases for the new functionality.~

### Description

In this PR we resolve issues regarding the participation websocket service implemented in https://github.com/ls1intum/ArTEMiS/pull/281.

Part of this PR:
 * Prevent the 'Resume' button from appearing for modeling and text exercises that resulted in an error.
 * Clear participation websockets when logging out 
 * Load results for mdoeling submissions using the participation websocket service
 * Fix missing instructor action buttons
 * Check if exercise ids match upon receiving a new participation in quiz component
 * Minor changes: cleanup, renaming, documentation, null checks

### CODE REVIEW

File: src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts [hunk 1]
```diff
@@ -65,16 +69,14 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
...
-                this.exercise.participations = cachedParticipations;
+                this.exercise.participations = cachedParticipations.filter((participation: Participation) => participation.student.id === this.currentUser.id);
                 this.mergeResultsAndSubmissionsForParticipations();
                 this.exerciseCategories = this.exerciseService.convertExerciseCategoriesFromServer(this.exercise);
                 this.subscribeForNewResults();
             });
         } else {
-            this.exerciseService.findResultsForExercise(this.exerciseId).subscribe((exercise: Exercise) => {
-                this.exercise = exercise;
-                this.exercise.isAtLeastTutor = this.accountService.isAtLeastTutorInCourse(this.exercise.course);
```
Comment (maximilianmeier) [inline src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts]: why die we remove the isAtLeastTutor property?

File: src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts [hunk 2]
```diff
@@ -65,16 +69,14 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
         if (cachedParticipations && cachedParticipations.length > 0) {
             this.exerciseService.find(this.exerciseId).subscribe((exerciseResponse: HttpResponse<Exercise>) => {
                 this.exercise = exerciseResponse.body;
-                this.exercise.isAtLeastTutor = this.accountService.isAtLeastTutorInCourse(this.exercise.course);
```
Comment (maximilianmeier) [inline src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts]: why die we remove the isAtLeastTutor property?

File: src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts (hunk 1 above)
Comment (SiggZ) [reply src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts]: It is already done in `checkPermission()` in the `ExerciseService`. We additionally have to set the `isAtLeastInstructor` property which is also done in the `checkPermission()` method.

File: src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts (hunk 2 above)
Comment (SiggZ) [reply src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts]: It is already done in `checkPermission()` in the `ExerciseService`. We additionally have to set the `isAtLeastInstructor` property which is also done in the `checkPermission()` method.

**Pull request merged**